core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -439,6 +441,23 @@ class KafkaController(val config: KafkaConfig,
     registerBrokerModificationsHandler(newBrokers)
   }
 
+  /**
+   * This method checks if a broker can be assigned partitions missing replicas. If so it triggers a reassignment
+   * replacing a placeholder broker with this new broker.
+   */
+  private def handlePartitionsMissingReplicas(newBrokers: Seq[Int]): Unit = {
+    for (brokerId <- newBrokers) {
+      val partitionsMissingReplicas = controllerContext.partitionsMissingReplicas(brokerId)
+      if (!partitionsMissingReplicas.isEmpty) {
+        for ((tp, replicas) <- partitionsMissingReplicas) {
+          val toRemove = Seq(replicas.filter(id => ReplicaAssignment.isPlaceholder(id)).head)
+          val replicaAssignment = ReplicaAssignment(replicas ++ Seq(brokerId).distinct, Seq(brokerId), toRemove)
```

Comment:
`Seq(brokerId).distinct`? Should `distinct` be done on the result of the ++ instead?